### DISCUSSION TITLE
Library/component to show country flags using VF

### DISCUSSION THREAD

**Discussion by sandykadam** [visual-framework/vf-core#1404] [Proposals]
There is requirement in Covid19portal to display the country flags, so it will be good to have some component to display flag based on country code.

**Comment by khawkins98** [comment]
Thanks @sandykadam 

I've added this to the next VF-UX catchup topics with the following questions:

> - Is this a good idea, would it be better to just write it out or use country codes? (DE, UK, IT, FR, USA, etc)
> - If we do want to use icons, do we already have them in our icon package or do we use a “generic” SVG source?
> - I expect the icons would have to be colour
> - I believe we still don’t have an official policy on “which countries” (Taiwan and similar contested names)

If you or @gabsie have any mockups,  they'd be helpful.

**Reply by sandykadam** [reply]
Please find attached proposed mockup, I have cropped it as its public forum to share :) 

![National Data Portals_cropped](https://user-images.githubusercontent.com/1227241/109830261-3b4e5680-7c36-11eb-9fa2-97889edd7192.png)

**Comment by sturobson** [comment]
Looking at the size in the proposed mock-up - I don't think these are really 'icons' as they're quite large. It also makes me think that we need an `avatar` style component that can incorporate something like this.

Aother thing is - the names of the country are listed alongside the flag -- so does this need the flag too?

A mindset of the visual framework is to have the text do the work where possible. So not having a calendar icon next to a set of dates, for example. 

Having it like this is almost like we are duplicating content.

**Reply by sturobson** [reply]
as an aside: 

This could make use of the `vf-avatar` component (that's to be created) for when you want initials, profile images, logos, flags, etc) and an updated `vf-summary` where you can include. 

This would allow teams to 'bring their own content' and have the consistency in the code behind it.
